proxy/envoy/ingress.go

```diff
@@ -161,17 +159,19 @@ func generateIngress(conf *IngressConfig) *Config {
 		},
 	}
 
-	// configure for HTTPS if provided with a secret name
-	if conf.Secret != "" {
-		// configure Envoy
+	// configure for HTTPS if there is a secret
+	if secret, err := conf.Secrets.GetSecret("*"); err != nil {
+		glog.Warningf("Error retrieving ingress secret: %v", err)
+	} else if secret != nil {
+		// configure Envoy for HTTPS
 		listener.Address = fmt.Sprintf("tcp://%s:443", WildcardAddress)
```

Comment:
I'm looking for a way to distinguish between HTTP vs HTTPS route rules. `L4MatchAttributes` doesn't seem to provide that:
```protobuf
/// L4 connection match attributes. Note that L4 connection matching
/// support is incomplete.
message L4MatchAttributes {
  /// IPv4 or IPv6 ip address with optional subnet. E.g., a.b.c.d/xx form or
  /// just a.b.c.d
  repeated string source_subnet = 1;

  /// IPv4 or IPv6 ip address of destination with optional subnet.
  /// E.g., a.b.c.d/xx form or just a.b.c.d. This is only valid when the destination
  /// service has several IPs and the application explicitly specifies a particular IP.
  repeated string destination_subnet = 2;
}
```